src/storage_eeprom.c

```diff
@@ -25,6 +25,8 @@ LOG_MODULE_REGISTER(thingset_storage_eeprom, CONFIG_THINGSET_SDK_LOG_LEVEL);
 #define EEPROM_DEVICE_NODE DT_NODELABEL(eeprom)
 #endif
 
+#define MAX_READ_SIZE 128
```

Comment:
128 was chosen to avoid exceeding the max chunk length of the I2C. Is that a suitable explanation? I wasn't entirely sure because not all EEPROMs will necessarily have the same issue.

Reply:
Suggestion for the comment:

Testing showed that reading from the EEPROM resulted in corrupted data under certain conditions if large amounts of data (>8 kB) were read at once. The STM32 I2C driver has an internal buffer size of 256 bytes, so the chunk size should be smaller than that.

Reply:
Sounds good, will change now!